### PR TITLE
Add hdoc support

### PR DESCRIPTION
**Pull request by no92**
Counterpart to managarm/bootstrap-managarm#255.

### CODE REVIEW

File: .hdoc.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[project]
+name = "mlibc"
+use_system_includes = false
+git_repo_url = "https://github.com/managarm/mlibc/"
+git_default_branch = "master"
+
+[paths]
+compile_commands = "../../build/pkg-builds/mlibc/compile_commands.json"
```
Comment (ArsenArsen) [inline .hdoc.toml:8]: ?

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
please add build system integration (including generating this file)

**Review by ElectrodeYT — APPROVED** [review]
LGTM

File: scripts/hdoc-config.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+# usage: [output hdoc configuration] [path to compile_commands.json]
+
+if [ ! "$#" -eq 2 ]; then
+	echo "usage: [output hdoc configuration] [build directory]" >&2
+	exit 1
+fi
+
+cp "scripts/hdoc.toml.template" "$1"
+echo "output_dir = \"$2/docs/\"" >> "$1"
+echo "compile_commands = \"$2/compile_commands.json\"" >> "$1"
```
Comment (ArsenArsen) [inline scripts/hdoc-config.sh:11]: https://mesonbuild.com/Reference-manual_functions.html#configure_file

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
no need for messy hacks.. just configure the file and add a target that runs hdoc

File: meson.build [hunk 3]
```diff
@@ -416,3 +416,19 @@ endif
...
+conf_data = configuration_data()
+conf_data.set('source_root', meson.source_root())
+conf_data.set('build_root', meson.build_root())
+configure_file(input: 'scripts/hdoc.toml.template',
+	output: '.hdoc.toml', configuration: conf_data)
+
+if hdoc.found()
+	run_target('hdoc',
+		command : ['sh', '-c',
+			'exec ' + hdoc.full_path() + ' --verbose'
```
Comment (ArsenArsen) [inline meson.build:431]: why?

**Review by ArsenArsen — APPROVED** [review]
pre-configuration files are usually suffixed .in, other than that, lgtm

**Review by Geertiebear — COMMENTED** [review]
Yeah, if the ``.template`` could be changed to ``.in`` then this is fine by me. Thanks for adding this, should provide a nice way to start writing some detailed docs for mlibc!

**Comment by no92** [comment]
Renamed the file to use the `.in` suffix.

**Pull request merged**